extensions/stats/plugin.cc

```diff
@@ -358,6 +358,14 @@ bool PluginRootContext::initializeDimensions(const json& j) {
           if (!name.empty() && name != factory_it.first) {
             continue;
           }
+
+          bool skip_reporting =
+              JsonGetField<bool>(metric, "skip_reporting").value_or(false);
+          if (skip_reporting) {
+            factories.erase(factory_it.first);
```

Comment:
This is likely problematic, Wasm test failed because of invalid memory access. Better record the factory to be removed and remove it after the iteration.

Reply:
Actually modifying inside iteration is what I was looking at when I thought we could do it at factory construction time. 
Are you saying that it is not possible because overrides will get lost?
I don’t see things added into factories anywhere else.

Reply:
New factory might be added at line 315.

Reply:
```
        for (auto factory_it =factories.begin(); factory_it != factories.end(); /*do not advance iterator here */) {

          if (skip_reporting) {
            // erase returns the increased iterator
            factory_it = factories.erase(factory_it /* erase iterator to avoid another lookup */);
            continue;
          }
       }
```
Most importantly, we need to advance iterator at other continue or end of the loop.

Reply:
ACK. I was planning on updating once i'd run the full test suite. I've adopted @lambdai 's suggested changes here.